perennialAutodidact/memory-snap
# 40 add loading state for photo api call

## Description of changes
- Added an action in the photos context to set the loading status of the API call
- Added an action in the photos context to set an error if the API call fails
- Created a loading spinner component which is rendered when the API call is loading
- Conditionally rendered the spinner component when the photos are loading and the Game component when they're loaded

## Test steps
- Reload the app. A loading spinner should briefly show before the game is rendered
- There isn't much to check with the error logic at the moment aside from the tests that are written for them in the code
